wordpress-mobile/WordPress-iOS
# Swift App Delegate 1: Notifications and Extensions

In an effort to get the ball rolling on The Great Swiftening (Swiftification?), let's start porting things over into extensions. First up, the app delegate. This PR moves over the methods for configuring and handling notification observers, and the app extensions.

There are more `@objc`s than I'd like, but we should be able to remove those when the rest of the app delegate is ported over.

To test:

There are quite a few things here, but they shouldn't take long to run though...

* Check tests pass
* Build and run
* Add breakpoints in each of the notification handlers
* Trigger a low memory warning in the simulator and check the handler is called
* Log in and check the default account change handler is called
* Go to the home screen and check the Today Widget (3D Touch the app icon) is configured for the account you added. Go into Photos and check the sharing extension is configured.
* Go back into the app and log out. Check the default account change handler is called.
* Check the today widget and share extension are no longer configured.
* Finally, in the system Settings app, go to General > Larger Text and change the font size. Check that the content size handler is called.
